### PR TITLE
Include updated options in sync callbacks

### LINKED ISSUE #3478: Collection#create doesn't properly pass final options to success callback
In https://github.com/jashkenas/backbone/blob/master/backbone.js#L986 `Collection#create` wraps the `success` callback (in order to handle `wait: true`). But the wrapper does not take the third argument, `options` -- when it calls the wrapped callback it instead just sends it the options `Collection#create` received. This breaks any custom sync code that might be modifying the options further downstream (for instance, it breaks https://github.com/nilbus/Backbone.dualStorage, which adds `dirty: true` to the options if the sync to server failed but succeeded to localStorage instead).

I don't know if this type of signaling via `options` is supported, but I can't see why it would hurt for the wrapper callback here to actually take the options the callback was called with and pass those along, instead of short-circuiting with the original options. Would a PR for this change be acceptable?

### PR DESCRIPTION
**Pull request by platinumazure**
Better version of #3484. Fixes #3478.

### CODE REVIEW

File: test/model.js [hunk 1]
```diff
@@ -580,6 +618,25 @@
...
+      },
+      urlRoot: '/test'
+    });
+
+    var model = new SpecialSyncModel({ id: 'id' });
+
+    model.destroy({ success: onSuccess });
+    this.ajaxSettings.success();
+
+    function onSuccess(model, response, options) {
```
Comment (akre54) [inline test/model.js]: No named functions. Use `onSuccess = function() { ... }` or put this inline.

Comment (platinumazure) [inline test/model.js]: Not even in a test?

This is an asynchronous bit, but in order for the test to read more naturally, I always like to put assertions after actions. This technique takes advantage of function hoisting to allow this to work. `var onSuccess = function() { /* ... */ }` will not work-- the declaration is hoisted, but the assignment doesn't happen until the line is reached, so I would have to move that function up. I would never do this in actual code.

That said, if there's no flexibility forthcoming on this, I can certainly make some tweaks. I just wanted to explain my rationale first and make sure it was understood I didn't do it randomly and/or without knowing the potential dangers.

Comment (akre54) [inline test/model.js]: > Not even in a test?

Nah, we want the source code and its tests to have a consistent style.

> This technique takes advantage of function hoisting to allow this to work

Right. Just move the `onSuccess` above the `destroy` call or inline it in the `options` object. Function hoisting is a bad part of JS and serves no real point here.

Comment (platinumazure) [inline test/model.js]: As I tried to explain, the "real point" is readability in the test: arrange, act, assert, from top to bottom. But I will push commits to fix this.

**Comment by platinumazure** [comment]
@akre54 Better?

File: test/model.js (hunk 1 above)
Comment (akre54) [inline test/model.js]: No I understand what you're saying, I just disagree it reads easier ;)

Thanks for fixing that up.

**Comment by akre54** [comment]
Much. Thanks.

**Pull request merged**